eos_data_distribution/ndn/dbus/chunks.py

```diff
@@ -201,15 +222,17 @@ def _on_progress(self, proxy, name, first_segment, last_segment):
             self.current_segment += 1
 
         self.current_segment -= 1
-        if not self._emitted_complete and self._check_for_complete():
+        self._complete()
+
+    def _complete(self):
+        if not self._emitted_complete:
             self._emitted_complete = True
-            proxy.call_complete(name, callback=self._on_complete)
+            self._interface.call_complete(
+                str(self.name), callback=self._on_complete)
 
     def _request_interest_complete(self, interface, res, interest):
         logger.info('Consumer: request interest complete: %s, %s, %s', interface, res, interest)
         name, final_segment, fd_list = interface.call_request_interest_finish(res)
```

Comment:
Since you have removed the calls to `setName()` and `_set_final_segment()` here and we don't do anything with the return values, should something be changed in the dbus interface itself? It strikes me as odd.